app/src/util/tools.ts

```diff
@@ -361,3 +361,39 @@ export const formatToShortNumber = (number: string, decimals = 2): string => {
 
   return `${parseFloat(rNumber.toFixed(decimals))}${units[unitIndex]}`
 }
+
+export const isObjectEqual = (obj1?: any, obj2?: any): boolean => {
+  if (!obj1 && obj2) return false
+  if (!obj2 && obj1) return false
+  if (!obj1 && !obj2) return true
+  if (typeof obj1 !== typeof obj2) return false
+
+  if (typeof obj1 !== 'object' && !Array.isArray(obj1)) {
+    return obj1 === obj2
+  }
+
+  if (Array.isArray(obj1)) {
+    for (let index = 0; index < obj1.length; index += 1) {
+      if (!isObjectEqual(obj1[index], obj2[index])) return false
+    }
+    return true
+  }
+
+  const keys1 = Object.keys(obj1)
+  const keys2 = Object.keys(obj2)
+
+  if (keys1.length !== keys2.length) return false
+
+  for (let keyIndex = 0; keyIndex < keys1.length; keyIndex += 1) {
+    const key = keys1[keyIndex]
+    if (typeof obj1[key] !== typeof obj2[key]) return false
+    if (!isObjectEqual(obj1[key], obj2[key])) return false
+  }
+
+  return true
+}
```

Comment:
What's the source of this function? It seems to work properly, but I'd be more comfortable knowing that it's battle tested

Reply:
@KadenZipfel , I built this function myself and tested in many cases.

Reply:
I would say that we should find a battle tested version of this, but now that I think of it, in the very unlikely case that there's an exception which fails, the worst case scenario would be very minor so I think it's fine.